### PR TITLE
Srsi macd

### PR DESCRIPTION
**Pull request by talvasconcelos**
Renamed and tweaked a bit the strategy and a small fix to the SMA file.

### CODE REVIEW

File: lib/sma.js [hunk 1]
```diff
@@ -2,15 +2,13 @@ module.exports = function container (get, set, clear) {
...
-        prev_sma = sum / length
-      }
-      s.period[key] = prev_sma
+      let SMA = s.lookback
+        .slice(0, length)
+        .reduce((sum, cur) => {
+          return sum + cur
+        }, 0)
+
+      s.period[key] = SMA / lenght
```
Comment (fasheng) [inline lib/sma.js]: typo: lenght -> length

**Comment by fasheng** [comment]
Besides the typo, you could also rename the strategy name in backtester.js, too.
Just rename all `rsi_macd` to `srsi_macd`:
```
sed -i 's/rsi_macd/srsi_macd/g' backtester.js
```
https://github.com/carlos8f/zenbot/blob/master/scripts/auto_backtester/backtester.js#L75

**Comment by talvasconcelos** [comment]
can i fix the typo and re commit the pr? how can i fix it?
this happens a lot... screw the length

**Comment by fasheng** [comment]
Yes, just edit the file and `git add .; git commit --amend` to avoid another commit and `git push --force`.

**Comment by fasheng** [comment]
And you must edit `lib/_codemap.js` to export sma function.

File: lib/sma.js [hunk 2]
```diff
@@ -2,15 +2,13 @@ module.exports = function container (get, set, clear) {
...
-        s.lookback.slice(0, length).forEach(function (period) {
-          sum += period[source_key]
-        })
-        prev_sma = sum / length
-      }
-      s.period[key] = prev_sma
+      let SMA = s.lookback
+        .slice(0, length)
+        .reduce((sum, cur) => {
+          return sum + cur
```
Comment (fasheng) [inline lib/sma.js]: `return sum + cur` -> `return sum + cur[source_key]`

Comment (talvasconcelos) [reply lib/sma.js]: done

**Review by fasheng — APPROVED** [review]

**Review by DeviaVir — APPROVED** [review]
:+1:

**Pull request merged**